### ISSUE TITLE
[Core] Parameter for scaling/normalizing

### ISSUE THREAD

**Issue by tblume1992** [Nixtla/mlforecast#158]
### Description

Currently using different scaling or transformers such as boxcox or a standard scaler need to be supplied by the user but we may want to enforce a pattern around when the transformation occurs, for example transforming after differencing will place the values in different magnitudes than if you transform first. Also it would be nice if the transforms were numba based!

### Use case

It would be great if there were a list of featured scalers we could pass as a string to the fit method or class creation such as:

```
import lightgbm as lgb
from mlforecast import MLForecast
from mlforecast.utils import generate_daily_series


series = generate_daily_series(100, equal_ends=True, n_static_features=2, static_as_categorical=True)

series['my_id_col'] = series['unique_id'].copy()

mlf = MLForecast(
    models=lgb.LGBMRegressor(),
    freq='D',
    lags=[7],
    scaler='standard' #new parameter
)
mlf.fit(series)

preds = mlf.predict(12)
```

**Comment by jmoralez** [comment]
Hi @tblume1992, thanks for raising this. The interface we thought for this is the `target_transforms` argument, as described [here](https://nixtla.github.io/mlforecast/docs/end_to_end_walkthrough.html#target-transformations).

So for example you can apply BoxCox and then differencing with something like the following:

```python
from mlforecast import MLForecast
from mlforecast.target_transforms import BaseTargetTransform, Differences
from mlforecast.utils import generate_daily_series
from sklearn.preprocessing import PowerTransformer

class BoxCoxTargetTransform(BaseTargetTransform):
    def fit_transform(self, df):
        df = df.copy()
        self.transformer_ = PowerTransformer(method='box-cox')
        df[self.target_col] = self.transformer_.fit_transform(df[[self.target_col]])[:, 0]
        return df
    
    def inverse_transform(self, df):
        df = df.copy()
        for col in df.columns.drop([self.id_col, self.time_col]):
            df[col] = self.transformer_.inverse_transform(df[[col]])
        return df

series = generate_daily_series(10)
fcst = MLForecast(
    models=[],
    freq='D',
    lags=[1, 2],
    target_transforms=[BoxCoxTargetTransform(), Differences([1])]
)
fcst.preprocess(series)
```

We've also thought about adding some high performance built-in transformations, we'll probably add some soon.

Please let us know if this helps.

**Comment by jmoralez** [comment]
Hey @tblume1992, do you want the Box-Cox transformation to be applied per serie?

**Comment by tblume1992** [comment]
> Hey @tblume1992, do you want the Box-Cox transformation to be applied per serie?

Yeah each transformation should be applied at the series level, things like a standard scaler, minmax etc would be great! A log transform would be fine globally. 

Box cox will be most impactful usually but the others can be good. Box cox can cause nans in the forecast so there is a 'robust' boxcox procedure you could add as well that uses a minmax scaler first and if there is a nan you replace with the min/max. Hopefully that makes sense!

**Comment by jmoralez** [comment]
I see. We recently implemented the standard scaler, you can find an example [here](https://nixtla.github.io/mlforecast/docs/target_transforms_guide.html#localstandardscaler). We'll work on adding more in the near future.

**Comment by jmoralez** [comment]
Hey @tblume1992, we just added the MinMax and Robust scalers as well as the BoxCox transformation as target transformations. Please let us know if we can close this issue.

**Comment by tblume1992** [comment]
Awesome thanks for the addition!
